.guix/modules/mahogany-package.scm

```diff
@@ -0,0 +1,165 @@
+;;; Mahogany --- StumpWM-like window manager written for Wayland
+;;; Copyright © 2023 Raven Hallsby <karl@hallsby.com>
+;;;
+;;; This file is part of Mahogany.
+;;;
+
+;;; Commentary:
+;;
+;; GNU Guix development package.  To build and install, run:
+;;
+;;   guix package -f guix.scm
+;;
+;; To use as the basis for a development environment, run:
+;;
+;;   guix shell -D -f guix.scm
+;;
+;;; Code:
+
+(define-module (mahogany-package)
+  #:use-module (guix packages)
+  #:use-module (guix gexp)
+  #:use-module (guix git-download)
+  #:use-module (guix build-system meson)
+  #:use-module (guix build-system asdf)
+  #:use-module (guix utils)
+  #:use-module ((guix licenses) #:prefix license:)
+  #:use-module (gnu packages autotools)
+  #:use-module (gnu packages pkg-config)
+  #:use-module (gnu packages build-tools)
+  #:use-module (gnu packages freedesktop)
+  #:use-module (gnu packages lisp-check)
+  #:use-module (gnu packages lisp-check)
+  #:use-module (gnu packages lisp-xyz)
+  #:use-module (gnu packages texinfo)
+  #:use-module (gnu packages xdisorg)
+  #:use-module (gnu packages wm))
+
+(define vcs-file?
+  ;; Return true if the given file is under version control.
+  (or (git-predicate (string-append (current-source-directory) "/../.."))
+      (const #t)))                                ;not in a Git checkout
+
+(define-public mahogany
+  (package
+    (name "mahogany")
+    (version (git-version "0.0.0" "0" "000000000000000000000000000000000000000000"))
+    (source (local-file "../.." "mahogany-checkout"
+                        #:recursive? #t
```

Comment:
If the recursive modules are deleted as part of the build process, why download them in the first place?